cgi-bin/DW/Media.pm

```diff
@@ -110,9 +110,15 @@ sub upload_media {
 
     # set the security
     my $sec = $opts{security} || 'public';
+    $sec = 'usemask' if $sec =~ /^(?:friends|access)$/;
```

Comment:
In that case I strongly feel it best to allow the defined or, since usemask security + no allowmask is easy to do by accident and will cause problems.  That can be in addition to the explicit setting paradigm you noted above.